api/v1alpha1/validation.go

```diff
@@ -684,6 +711,13 @@ func validateAppProvider(app ApplicationProviderSpec, appType ApplicationProvide
 			errs = append(errs, fmt.Errorf("image reference cannot be empty when application name is not provided"))
 		}
 		errs = append(errs, validation.ValidateOciImageReference(&provider.Image, "spec.applications[].image")...)
+	case InlineApplicationProviderType:
```

Comment:
Looks like there's some missing validation for Inline apps.
If I attempt to PATCH a device to add an inline application, I get 400 request
with the message `Error 400: Bad Request: unsupported application provider type: inline`, and it seems to be coming from https://github.com/flightctl/flightctl/blob/5fb9d71868534401e54f18b8245978331a19b349/api/v1alpha1/validation.go#L762

Patch body is
```
[{"path":"/spec/applications","op":"replace","value":[{"type":"compose","inline":{"encoding":"base64","data":{"Chart.yaml":"YXBwTmFtZTogbXktYXBwCmFwcFR5cGU6IHdlYgo="}}}]}]
```

In this regard, I imagine `name` should be required for inline applications, while we may want to keep it optional for `ImageApplicationProviderType`?

Reply:
It may be because you're still working on this PR, but FYI there's another missing validation regarding inline applications
https://github.com/flightctl/flightctl/blob/5fb9d71868534401e54f18b8245978331a19b349/internal/tasks/device_render.go#L260

Reply:
thanks for the heads up, yeah I am working on the updates and will make sure these are all covered

Reply:
i stubbed out the support and it will be implemented on the next pr